recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+    name = "mimalloc"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/mimalloc"
+    description = "mimalloc is a compact general purpose allocator with excellent performance."
+    topics = ("libraries", "c", "c++", "malloc", "allocator")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False],
+               "override": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False,
+                       "override": True}
+    generators = "cmake", "visual_studio"
+    exports_sources = "patches/**"
+    _cmake_i = None
+    _msbuild_i = None
+
+    @property
+    def _cmake_install_folder(self):
+        p1 = self.version.find(".")
+        if p1 != -1:
+            p2 = self.version.find(".", p1 + 1)
+            if p2 != -1:
+                return self.name + "-" + self.version[0:p2]
+            else:
+                return self.name + "-" + self.version
+        else:
+            return self.name + "-" + self.version
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _cmake(self):
+        if not self._cmake_i:
+            self._cmake_i = CMake(self)
+            self._cmake_i.definitions["MI_BUILD_TESTS"] = False
+            if self.options.shared:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = True
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = False
+            else:
+                self._cmake_i.definitions["MI_BUILD_SHARED"] = False
+                self._cmake_i.definitions["MI_BUILD_STATIC"] = True
+            self._cmake_i.configure(source_folder=self._source_subfolder,
+                build_folder=self._build_subfolder)
+        return self._cmake_i
+
+    @property
+    def _msbuild(self):
+        if not self._msbuild_i:
+            self._msbuild_i = MSBuild(self)
+        return self._msbuild_i
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("mimalloc-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        pass
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, "14")
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            if Version(self.settings.compiler.version) < "15":
+                raise ConanInvalidConfiguration("Visual Studio 15 2017 or newer is required")
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+        if self.settings.compiler == "Visual Studio":
```

Comment:
Doesn't the cmake script work for Visual Studio?
I guess it does, mimalloc being developed by Microsoft.

Reply:
Normalized to CMake builds. Cf. b028ffa